tests/rkt_tests.go

```diff
@@ -118,21 +118,36 @@ func spawnOrFail(t *testing.T, cmd string) *gexpect.ExpectSubprocess {
 	return child
 }
 
-func waitOrFail(t *testing.T, child *gexpect.ExpectSubprocess, shouldSucceed bool) {
+func getExitStatus(err error) int {
+	if err == nil {
+		return 0
+	}
+	if exiterr, ok := err.(*exec.ExitError); ok {
+		// the program has exited with an exit code != 0
+		if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+			return status.ExitStatus()
+		}
+	}
+	return -1
+}
+
+func waitOrFail(t *testing.T, child *gexpect.ExpectSubprocess, expectedStatus int) {
 	err := child.Wait()
+	status := getExitStatus(err)
+	if status != expectedStatus {
+		t.Fatalf("rkt terminated with unexpected status %d, expected %d\nOutput:\n%s", status, expectedStatus, child.Collect())
+	}
 	switch {
-	case !shouldSucceed && err == nil:
+	case expectedStatus != 0 && err == nil:
 		t.Fatalf("Expected test to fail but it didn't\nOutput:\n%s", child.Collect())
-	case shouldSucceed && err != nil:
+	case expectedStatus == 0 && err != nil:
```

Comment:
It seems the two cases of the `switch` will be caught in the test `status != expectedStatus` before.